File: Sorting_2/quickSort/quickSort.js
const quickSort =(arr,low,high)=>{
    if(low<high){
        let pIndex=partition(arr,low,high);
        quickSort(arr,low,pIndex-1);
        quickSort(arr,pIndex+1,high);

    }

}
const partition=(arr,low,high)=>{
    let pivot=arr[low];
    let i=low;
    let j=high;
    while(i<j){
        while(arr[low]<=pivot && i<=high-1){
            i++;
        }
        while(arr[j]>pivot && j>=low+1){
            j--;
        }
        if(i<j){
            [arr[i],arr[j]]=[arr[j],arr[i]];
        }
    }
    [arr[low],arr[j]]=[arr[j],arr[low]];
    return j;
}
let a =[76,8,9,6,4,2];
quickSort(a,0,a.length-1);
console.log(a.join(" "));

File: Sorting_2/quickSort/quickSort.java
package Sorting_2.quickSort;

public class quickSort {
    public static void main(String[] args) {
        int[] a = { 87, 90, 12, 45, 76 };
        quickSort(a, 0, a.length - 1);
        for (int i = 0; i < a.length; i++) {
            System.out.print(a[i] + " ");
        }
        ;

    }

    public static void quickSort(int[] arr, int low, int high) {
        
        if(low<high){
            int pIndex=partition(arr,low,high);
            quickSort(arr,low,pIndex-1);
            quickSort(arr,pIndex+1,high);
        }

    }
    public static int partition(int[] arr, int low,int high){
        int pivot=arr[low];
        int i=low;
        int j=high;
        while(i<j){
             while(arr[i]<=pivot && i<=high-1){
            i++;
        }
        while(arr[j]>pivot && j>=low+1){
            j--;
        }
        if(i<j){
            int temp=arr[i];
            arr[i]=arr[j];
            arr[j]=temp;
        }

        }
        int temp=arr[j];
        arr[j]=arr[low];
        arr[low]=temp;
        return j;
       

    }

}
